simplecpp.cpp

```diff
@@ -1282,14 +1270,97 @@ void simplecpp::TokenList::constFoldBitwise(Token *tok)
                 result = (stringToLL(tok->previous->str()) ^ stringToLL(tok->next->str()));
             else /*if (*op == '|')*/
                 result = (stringToLL(tok->previous->str()) | stringToLL(tok->next->str()));
-            tok = tok->previous;
-            tok->setstr(toString(result));
-            deleteToken(tok->next);
-            deleteToken(tok->next);
+            simpleSquash(tok, toString(result));
         }
     }
 }
 
+static simplecpp::Token * stepForward(simplecpp::Token *tok)
+{
+    if (!tok)
+        return nullptr;
+    return tok->next;
+}
+
+static simplecpp::Token * stepBack(simplecpp::Token *tok)
+{
+    if (!tok)
+        return nullptr;
+    return tok->previous;
+}
+
+void simplecpp::TokenList::simpleSquash(Token *&tok, const std::string & result)
+{
+    tok = tok->previous;
+    tok->setstr(result);
+    deleteToken(tok->next);
+    deleteToken(tok->next);
+}
+
+void simplecpp::TokenList::squashTokens(Token *&tok, const std::set<std::string> & breakPoints, bool forwardDirection, const std::string & result)
+{
+    simplecpp::Token *(*step)(simplecpp::Token *) = &stepForward;
```

Comment:
It's not really required to create stepForward and stepBack intermediate functions

```suggestion
    Token* (Token:: * const step)() = forwardDirection ? &Token::next : &Token::previous;
```
To use `step` you'd need to use the `->*` operator though: `(tok->*step)()`.

Reply:
But those are not functions